document-store/src/integrationTest/java/org/hypertrace/core/documentstore/mongo/MongoDocStoreTest.java

```diff
@@ -1102,4 +1106,68 @@ public void test_bulkOperationOnArrayValue_setOperation() throws Exception {
       assertEquals(expectedAttributesJsonNode, attributesJsonNode);
     }
   }
+
+  @Test
+  public void test_unnest() throws IOException {
+    datastore.createCollection(COLLECTION_NAME, null);
+    Collection collection = datastore.getCollection(COLLECTION_NAME);
+
+    collection.upsert(
+        new SingleValueKey("default", "testKey1"),
+        Utils.createDocument(
+            ImmutablePair.of("score", 20),
+            ImmutablePair.of("field", "x1"),
+            ImmutablePair.of(
+                "arrayField",
+                List.of(
+                    Map.of("nestedArrayField", List.of("c1", "c2")),
+                    Map.of("nestedArrayField", List.of("c1", "c7"))))));
+
+    collection.upsert(
+        new SingleValueKey("default", "testKey2"),
+        Utils.createDocument(
+            ImmutablePair.of("score", 30),
+            ImmutablePair.of("field", "x1"),
+            ImmutablePair.of(
+                "arrayField", List.of(Map.of("nestedArrayField", List.of("c4", "c2"))))));
+
+    org.hypertrace.core.documentstore.query.Query query =
+        org.hypertrace.core.documentstore.query.Query.builder()
+            .addSelection(IdentifierExpression.of("score"))
+            .addSelection(IdentifierExpression.of("arrayField.nestedArrayField"))
+            .addAggregation(IdentifierExpression.of("arrayField.nestedArrayField"))
+            .addSelection(AggregateExpression.of(SUM, IdentifierExpression.of("score")), "total")
+            .addFromClause(UnnestExpression.of(IdentifierExpression.of("arrayField")))
+            .addFromClause(
+                UnnestExpression.of(IdentifierExpression.of("arrayField.nestedArrayField")))
+            .build();
+
+    Iterator<Document> iterator = collection.aggregate(query);
+
+    List<JsonNode> jsonNodes = new ArrayList<>();
```

Comment:
By moving this test to MongoQueryExecutorIntegrationTest, the assertion part can be simplified by invoking `MongoQueryExecutorIntegrationTest.assertDocsEqual()`.

Reply:
You may need to apply sorting for verifying.